### ISSUE TITLE
REST API rate limit?

### ISSUE THREAD

**Issue by xme** [MISP/MISP#1176]
I don't think that the REST API has a system to limit calls (per IP, per min, ...).
Open question:
In case of an "open" MISP instance, what could you recommend to control API calls (except completely disabling them :-)

/x

**Comment by iglocska** [comment]
Interesting. Flat out limiting it could be dangerous as you'd seriously hamstring the sync mechanism, but this could be interesting as a setting for API users. I wonder what would be the best way to go for this.

**Comment by xme** [comment]
MISP knows its peers... They could just be excluded from any limitation to not degrade sync operations.

**Comment by iglocska** [comment]
Not necessarily, you can have a partner that pulls from you / pushes to you without you knowing about his/her instance as long as you give them a sync user.

**Comment by xme** [comment]
Hmmm.... Too easy :)

**Comment by adulau** [comment]
We have now a limit of memory usage and time execution per role. A specific role could be assigned with limited resource in order to avoid such issue. What do you think?

**Comment by github-germ** [comment]
This sounds very interesting. Can you provide some details on the settings along with a few use cases illustrating how applying these settings will limit specific usage examples? Thx.

**Comment by adulau** [comment]
It's part of the role (check the highlighted part) :
![re-bis](https://user-images.githubusercontent.com/3309/50482808-e3c65780-09e8-11e9-839f-cbfcdf8570bf.png)
